### PR TITLE
spec: Add user defined port flag

### PR DESCRIPTION
**Pull request by exdx**
Adds a new flag to specify an arbitrary port. Defaults to 9090.

Signed-off-by: Dan Sover <dan.sover@avalabs.org>

### CODE REVIEW

File: avalanche-ops/src/aws/cfn-templates/vpc.yaml [hunk 1]
```diff
@@ -56,6 +56,11 @@ Parameters:
     Type: Number
     Default: 9650
     Description: HTTP port
+  
+  UserProvidedPort:
```
Comment (gyuho) [inline avalanche-ops/src/aws/cfn-templates/vpc.yaml:60]: This is not being used below? We need to create the security group ingress rule, as we do for SSH.

**Comment by gyuho** [comment]
Can we also define cidr range for this port? Default to 0.0.0.0 and set the same default value as `SshPortIngressIpv4Range` where we resolve the public IP from your local laptop? We should also make the cidr range for this configurable.

File: avalanche-ops/src/aws/cfn-templates/vpc.yaml [hunk 2]
```diff
@@ -273,6 +285,15 @@ Resources:
       FromPort: !Ref HttpPort
       ToPort: !Ref HttpPort
       CidrIp: !Ref HttpPortIngressIpv4Range
+  
+  UserDefinedIngress:
+    Type: AWS::EC2::SecurityGroupIngress
+    Properties:
+      GroupId: !Ref SecurityGroup
```
Comment (gyuho) [inline avalanche-ops/src/aws/cfn-templates/vpc.yaml:292]: We need to define a "new" security group. The current changes opens the port to the "node", which we don't want. And use the new security group (sg) here

File: avalanche-ops/src/aws/cfn-templates/vpc.yaml [hunk 3]
```diff
@@ -52,10 +52,22 @@ Parameters:
     AllowedPattern: '((\d{1,3})\.){3}\d{1,3}/\d{1,2}'
     Description: IP range for HTTP inbound traffic
 
+  # limit this to your IP
+  UserDefinedPortIngressIpv4Range:
```
Comment (gyuho) [inline avalanche-ops/src/aws/cfn-templates/vpc.yaml:56]: Move this above, `UserDefinedPort`?

File: avalanche-ops/src/aws/cfn-templates/vpc.yaml [hunk 4]
```diff
@@ -273,6 +285,15 @@ Resources:
...
       CidrIp: !Ref HttpPortIngressIpv4Range
+  
+  UserDefinedIngress:
+    Type: AWS::EC2::SecurityGroupIngress
+    Properties:
+      GroupId: !Ref SecurityGroup
+      IpProtocol: tcp
+      FromPort: !Ref UserDefinedPort
+      ToPort: !Ref UserDefinedPort
+      CidrIp: !Ref SshPortIngressIpv4Range
```
Comment (gyuho) [inline avalanche-ops/src/aws/cfn-templates/vpc.yaml:296]: UserDefinedPortIngressIpv4Range?

**Review by gyuho — DISMISSED** [review]
We need to define a separate sg for dev machine, and expose that as outputs, and use that output in the dev machine asg creation.

**Comment by exdx** [comment]
> use that output in the dev machine asg creation

Cool. I'm working on this last step now.

File: avalancheup-aws/src/apply/mod.rs [hunk 5]
```diff
@@ -2726,14 +2738,32 @@ default-spec --log-level=info --funded-keys={funded_keys} --region={region} --up
...
 
         let mut cfn_params = Vec::new();
         for (k, v) in regional_common_dev_machine_asg_params.iter() {
             log::info!("dev-machine CFN parameter: '{k}'='{v}'");
             cfn_params.push(build_param(k, v));
         }
 
+        // Add the security group ids to the dev machine ASG
+        cfn_params.push(build_param(
+            "NetworkInterfaces.0.Groups.0",
```
Comment (gyuho) [inline avalancheup-aws/src/apply/mod.rs:2752]: Does this work? Just using `DevSecurityGroupId` parameter wouldn't work?

Comment (exdx) [reply avalancheup-aws/src/apply/mod.rs]: I think you're right. We already defined the `SecurityGroupId` so updating the dev one should be fine in the `!Join`. Updated. I think this is ready

**Comment by exdx** [comment]
```
error[E0635]: unknown feature `proc_macro_span_shrink`
  --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/proc-macro2-1.0.59/src/lib.rs:92:30
   |
92 |     feature(proc_macro_span, proc_macro_span_shrink)
```

from checking cargo unused job. But I believe that's unrelated to this PR

**Pull request merged**